torch/quantization/observer.py

```diff
@@ -0,0 +1,61 @@
+from __future__ import absolute_import, division, print_function, unicode_literals
+import torch.nn as nn
+import torch
+from functools import partial
+
+class Observer(nn.Module):
+    r"""Default Observer Module
+    A default implementation of the observer module, only works for
+    per_tensor_affine quantization scheme.
+    The module will record the running average of max and min value of the
+    observed Tensor and calulate_qparams will calculate the scale and zero_point
+
+    Other types of Observers should follow the same API, the first argument of
+    the constructor should be qoptions, and it can take arbitrary number of
+    arguments after that. and it should provide a calculate_qparam function.
+    TODO: Maybe add an abstract Observer class?
+    """
+    def __init__(self, **kwargs):
```

Comment:
why do you have this weird kwargs business? regular arguments with defgault values? (and docs please!)

Reply:
reason for kwargs is this is a base class and other types of observers might have some extra arguments. but I think we should have some default arguments like dtype and qscheme as well.

Reply:
wait, how base class is related here. If one of the descendants overrides this constructor, it'd pass the subset of argument down. There's no need for kwargs

Reply:
OK